atomsci/ddm/pipeline/model_datasets.py

```diff
@@ -655,11 +661,31 @@ def combined_training_data(self):
         # All of the splits have the same combined train/valid data, regardless of whether we're using
         # k-fold or train/valid/test splitting.
         if self.combined_train_valid_data is None:
+            # normally combining one fold is sufficient, but if SMOTE or undersampling is being used
+            # just combining the first fold isn't enough
```

Comment:
I don't think a compound can ever be wiped entirely out of existence due to undersampling. Undersampling is only applied to the [training set of each fold.](https://github.com/rwilfong/AMPL/blob/1.7.0/atomsci/ddm/pipeline/model_pipeline.py#L307)
And since every compound has a 'turn' in the validation set, that compound must appear at least once.

This isn't tested, but do we need to test it anywhere? I think it's ok if a compound is dropped entirely, since that's what happens when using undersampling without k-fold validation.